addons/point_of_sale/models/pos_order.py

```diff
@@ -652,12 +652,24 @@ def action_receipt_to_customer(self, name, client, ticket, order_ids=False):
         orders = self.browse(order_ids) if order_ids else self
 
         message = _("<p>Dear %s,<br/>Here is your electronic ticket for the %s. </p>") % (client['name'], name)
+
+        filename = 'Receipt-' + name + '.jpg'
+        receipt = self.env['ir.attachment'].create({
+            'name': filename,
+            'type': 'binary',
+            'datas': ticket,
+            'res_model': 'pos_order',
+            'res_id': order_ids[0],
+            'store_fname': filename,
+            'mimetype': 'image/jpeg',
+        })
         template_data = {
             'subject': _('Receipt %s') % name,
-            'body_html': message + '<img src="data:image/jpeg;base64,%s"/>' % ticket,
+            'body_html': message,
             'author_id': self.env.user.partner_id.id,
             'email_from': self.env.company.email or self.env.user.email_formatted,
-            'email_to': client['email']
+            'email_to': client['email'],
+            'attachment_ids': [receipt.id],
```

Comment:
shouldn't we use a list of orm commands [(4, receipt.id, 0)] instead of a list of id?

Reply:
Probably better, yes.